services/ingestion/event_subscriber.go

```diff
@@ -204,49 +206,108 @@ func (r *RPCEventSubscriber) backfill(ctx context.Context, height uint64) <-chan
 		}()
 
 		for {
-			// check if the current height is still in past sporks, and if not return since we are done with backfilling
-			if !r.client.IsPastSpork(height) {
+			// check if the current currentHeight is still in past sporks, and if not return since we are done with backfilling
+			if !r.client.IsPastSpork(currentHeight) {
 				r.logger.Info().
-					Uint64("height", height).
+					Uint64("height", currentHeight).
 					Msg("completed backfilling")
 
 				return
 			}
 
-			latestHeight, err := r.client.GetLatestHeightForSpork(ctx, height)
+			currentHeight, err := r.backfillSpork(ctx, currentHeight, eventsChan)
 			if err != nil {
+				r.logger.Error().Err(err).Msg("error backfilling spork")
 				eventsChan <- models.NewBlockEventsError(err)
 				return
 			}
 
 			r.logger.Info().
-				Uint64("start-height", height).
-				Uint64("last-spork-height", latestHeight).
-				Msg("backfilling spork")
+				Uint64("next-height", currentHeight).
+				Msg("reached the end of spork, checking next spork")
+		}
+	}()
 
-			for ev := range r.subscribe(ctx, height) {
-				eventsChan <- ev
+	return eventsChan
+}
 
-				if ev.Err != nil {
-					return
-				}
+// maxRangeForGetEvents is the maximum range of blocks that can be fetched using the GetEventsForHeightRange method.
+const maxRangeForGetEvents = uint64(249)
 
-				r.logger.Debug().Msg(fmt.Sprintf("backfilling [%d / %d]...", ev.Events.CadenceHeight(), latestHeight))
+func (r *RPCEventSubscriber) backfillSpork(ctx context.Context, fromHeight uint64, eventsChan chan<- models.BlockEvents) (uint64, error) {
```

Comment:
can you add comment to this function to explain the `Spork` in the name? and also the return value?

Is the returned height the first height for the next spork of the fromHeight?